client/src/BibleEngineClient.ts

```diff
@@ -22,6 +23,43 @@ export class BibleEngineClient {
         if (bibleEngineOptions) this.localBibleEngine = new BibleEngine(bibleEngineOptions);
     }
 
+    static chunk(arr: any, chunkSize = 1, cache: any[] = []) {
+        const tmp = [...arr];
+        if (chunkSize <= 0) return cache;
+        while (tmp.length) cache.push(tmp.splice(0, chunkSize));
+        return cache;
+    }
+
+    static async getBookIndexFile(versionUid: string, baseS3Url: string) {
+        const fileIndexUrl = `${baseS3Url}/${versionUid}/index.json`;
+        const response = await fetch(fileIndexUrl);
+        const fileIndex = await response.json();
+        return fileIndex;
+    }
+
+    async downloadVersion(versionUid: string, baseS3Url: string) {
```

Comment:
is there any reason to include `S3` in the variable name? if not, i'd suggest removing it to avoid confusion (since the json files can be hosted anywhere)

Reply:
removed all the download code, I'm going to do a followup PR with tests and the like